pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1531,6 +1535,12 @@ func (c *Cloud) CreateDisk(volumeOptions *VolumeOptions) (string, error) {
 	volSize := int64(volumeOptions.CapacityGB)
 	request.Size = &volSize
 	request.VolumeType = &createType
+	request.Encrypted = &volumeOptions.Encrypted
+	request.KmsKeyId = &volumeOptions.KmsKeyId
```

Comment:
I tend to prefer the `aws.String(s)` help function now that it is available (and Bool, Int64 etc friends).  But just my personal preference.

Have we tested this without a KmsKeyId?  Not all AWS APIs like being passed "" (which is why we have to do the whole crazy pointer dance in the first place!)